### PR TITLE
[Samples][Java] Fix disparities of 50.teams-messaging-extensions-search sample

### PR DESCRIPTION
**Pull request by Batta32**
Related to microsoft/botbuilder-java1165

## Proposed Changes
We compared the documentation/code migration/behavior of the `50.teams-messaging-extensions-search` sample between [Java](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/java_springboot/50.teams-messaging-extensions-search) and [C#](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/50.teams-messaging-extensions-search) and we found disparities and fixes that this PR includes.

- Add missing validations of properties
- Update README
- Add missing comments
- Clean-up autogenerated files/folder related to Eclipse (e.g. target/.classpath)
- Add `package-info.java` file

## Testing
_Teams Messaging Extensions Search sample working as expected_
image

### CODE REVIEW

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md [hunk 1]
```diff
@@ -17,35 +17,19 @@ build a Search-based Messaging Extension.
 > Note these instructions are for running the sample on your local machine, the tunnelling solution is required because
 the Teams service needs to call into the bot.
 
-1) Clone the repository
-
-    ```bash
-    git clone https://github.com/Microsoft/botbuilder-java.git
-    ```
-
-1) Run ngrok - point to port 3978
```
Comment (fran893) [inline samples/java_springboot/50.teams-messaging-extensions-search/README.md:26]: Why are you removing this step?

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md [hunk 2]
```diff
@@ -17,35 +17,19 @@ build a Search-based Messaging Extension.
...
-
-    ```bash
-    ngrok http -host-header=rewrite 3978
-    ```
-
-1) Create [Bot Framework registration resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-quickstart-registration) in Azure
+- Create [Bot Framework registration resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-quickstart-registration) in Azure
```
Comment (fran893) [inline samples/java_springboot/50.teams-messaging-extensions-search/README.md:21]: In this step I added a reference to the deploy section, I don't know if it is correct

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md (hunk 1 above)
Comment (Batta32) [reply samples/java_springboot/50.teams-messaging-extensions-search/README.md:26]: In order to have a consistency with the previous Java samples

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md (hunk 2 above)
Comment (Batta32) [reply samples/java_springboot/50.teams-messaging-extensions-search/README.md:21]: I'll check

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md (hunk 1 above)
Comment (Batta32) [reply samples/java_springboot/50.teams-messaging-extensions-search/README.md:26]: Ok, you are right, this should be added

File: samples/java_springboot/50.teams-messaging-extensions-search/README.md (hunk 2 above)
Comment (fran893) [reply samples/java_springboot/50.teams-messaging-extensions-search/README.md:21]: I did not committed yet, but just to let you know

Comment (Batta32) [reply samples/java_springboot/50.teams-messaging-extensions-search/README.md:21]: I don't think that we need to add a reference to the deploy here, I mean.. if we add this we should change the previous samples too.. maybe in the next round

**Review by fran893 — APPROVED** [review]

**Review by FedericoBernal — APPROVED** [review]
LGTM

**Pull request closed without merging**